src/Settings/Settings/SettingsLocation.php

```diff
@@ -0,0 +1,153 @@
+		];
+	}
+
+	/**
+	 * Get Form settings data array
+	 *
+	 * @param string $formId Form Id.
+	 *
+	 * @return array<int, array<string, mixed>>
+	 */
+	public function getSettingsData(string $formId): array
+	{
+		$output = [
+			[
+				'component' => 'intro',
+				'introTitle' => __('Location settings', 'eightshift-forms'),
+				'introSubtitle' => __('See all the locations where your block is assigned in one place. Keep in mind that this page can take a little while to open if you have a lot of content on your website.', 'eightshift-forms'),
+			],
+		];
+
+		$locations = $this->getBlockLocations($formId);
+
+		if (!$locations) {
+			$output[] = [
+				'component' => 'highlighted-content',
+				'highlightedContentTitle' => __('We are sorry but', 'eightshift-forms'),
+				'highlightedContentSubtitle' => __('It looks like your form hasn\'t been added to any page.', 'eightshift-forms'),
+			];
+		}
+
+		$output[] = [
+			'component' => 'admin-listing',
+			'adminListingForms' => $this->getBlockLocations($formId),
+		];
+
+		return $output;
+	}
+
+	/**
+	 * Get global settings array for building settings page.
+	 *
+	 * @return array<int, array<string, mixed>>
+	 */
+	public function getSettingsGlobalData(): array
+	{
+		return [];
+	}
+
+	/**
+	 * Return all posts where form is assigned.
+	 *
+	 * @param string $formId Form Id.
+	 *
+	 * @return array<int, mixed>
+	 */
+	private function getBlockLocations(string $formId): array
+	{
+		global $wpdb;
+
+		$items = $wpdb->get_results( // phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.DirectDatabaseQuery.NoCaching
+			$wpdb->prepare(
+				"SELECT *
```

Comment:
Since you are only using `ID`, `post_type`, `post_title`, and `post_status` from the results, you can specify these in the `SELECT` clause. Could shave a couple of milliseconds off your query (useful on sites with tons of content in the posts table).

I ran a similar query on my local db where I had 42k entries in the posts table and the query was 40ms faster 🙂

Reply:
awesome tnx a lot